crates/rpc/rpc-types-compat/src/engine/payload.rs

```diff
@@ -0,0 +1,343 @@
+//! Standalone functions for engine specific
+use reth_primitives::{
+    constants::{MAXIMUM_EXTRA_DATA_SIZE, MIN_PROTOCOL_BASE_FEE_U256},
+    proofs::{self, EMPTY_LIST_HASH},
+    Block, Header, SealedBlock, TransactionSigned, UintTryTo, Withdrawal, H256, U256,
+};
+use reth_rlp::Decodable;
+use reth_rpc_types::engine::{
+    payload::{
+        ExecutionPayloadBodyV1, ExecutionPayloadFieldV2, ExecutionPayloadInputV2,
+        StandaloneWithdraw,
+    },
+    ExecutionPayload, ExecutionPayloadV1, ExecutionPayloadV2, ExecutionPayloadV3, PayloadError,
+};
+
+/// Converts [ExecutionPayloadV1] to [Block]
+pub fn try_convert_from_execution_payload_v1_to_block(
```

Comment:
these functions are a bit too verbose imo

I think we can shorten all of these functions to just `try_*`